Update README.md with some extra details (notes)

nshumoogum opened the pull request:
### What

Includes some edge case requests that can and cannot be made as well as
description (notes) of some underlying updates to resources that were
not directly called to be updated (e.g. when a version gets published, the edition and dataset resources automatically get updated accordingly).

### How to review

Check for spelling and grammar mistakes and changes to README.md make sense.

### Who can review

Team A or B

nshumoogum committed:
Update README.md with some extra details (notes)

Includes some edge case requests that can and cannot be made as well as
description (notes) of some underlying updates to resources that were
not directly called to be updated.

CarlHembrough commented on README.md:
```diff
@@ -149,6 +160,13 @@ Example curl command to PUT dataset version state:
+Once the version has been updated with a state of `published`, then the edition is updated; this includes a change of state from `created` to `published` if the resource was not already published from a previous version and the `links.latest_version` is updated.
+Once the edition resource is successfully updated then the dataset resource is updated with a new state `published` on the next sub document and the next sub document is copied over to the current sub document.
+
+It is possible to add a `collection_id` before changing the state to `associated`, two separate requests.
```
I think this was copied and paste here by accident as its repeated above?

nshumoogum committed:
Remove duplicate paragraph across notes in README.md

CarlHembrough reviewed (approved):
👍

nshumoogum committed:
Fix typos

Mistake in example json (no need for comma) and state is `completed` NOT
`complete`
